### PR TITLE
rabbitmq module: Update documentation

### PR DESCRIPTION
**Pull request by roberth**
Elaborate on the two config file formats.

###### Motivation for this change

The module was changed to support both formats, but the documentation was quite minimal.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/amqp/rabbitmq.nix [hunk 1]
```diff
@@ -87,20 +87,37 @@ in {
...
         default = "";
         type = types.str;
         description = ''
-          Verbatim advanced configuration file contents.
-          Prefered way is to use configItems.
+          Verbatim advanced configuration file contents using the Erlang syntax.
+          This is also known as the <literal>advanced.config</literal> file or the old config format.
+
+          Where possible, the use <literal>configItems</literal> is preferred. However, nested
+          data structures can only be expressed properly using the <literal>configItems</literal> option.
```
Comment (lsix) [inline nixos/modules/services/amqp/rabbitmq.nix]: I think you meant `config` instead of `configItems` here

Comment (roberth) [reply nixos/modules/services/amqp/rabbitmq.nix]: Yes, thank you!

File: nixos/modules/services/amqp/rabbitmq.nix [hunk 2]
```diff
@@ -87,20 +87,37 @@ in {
...
-          See http://www.rabbitmq.com/configure.html
+          Configuration options in RabbitMQ's new config file format,
+          which is a simple key-value format that can not express nested
+          data structures. This is known as the <literal>rabbitmq.conf</literal> file,
+          although outside NixOS that filename may have Erlang syntax, particularly
+          prior to RabbitMQ 3.7.0.
+
+          If you do need to express nested data structures, you can use
+          <literal>config</literal> option. Configuration from <literal>config</literal>
+          will be merged into the these options by RabbitMQ at runtime to
```
Comment (c0bw3b) [inline nixos/modules/services/amqp/rabbitmq.nix]: Problem with `the these`

File: nixos/modules/services/amqp/rabbitmq.nix [hunk 3]
```diff
@@ -87,20 +87,37 @@ in {
...
       config = mkOption {
         default = "";
         type = types.str;
         description = ''
-          Verbatim advanced configuration file contents.
-          Prefered way is to use configItems.
+          Verbatim advanced configuration file contents using the Erlang syntax.
+          This is also known as the <literal>advanced.config</literal> file or the old config format.
+
+          Where possible, the use <literal>configItems</literal> is preferred. However, nested
```
Comment (c0bw3b) [inline nixos/modules/services/amqp/rabbitmq.nix]: "the use **of** configItems"  _or_  "using configItems"  _or just_ "configItems is preferred whenever possible"

**Review by c0bw3b — APPROVED** [review]

**Pull request merged**